runtime/utils/common/src/aws/smithy/kotlin/runtime/util/Platform.kt

```diff
@@ -5,45 +5,24 @@
 
 package aws.smithy.kotlin.runtime.util
 
-@InternalApi
-public expect object Platform {
-    /**
-     * Get an environment variable or null
-     *
-     * @param key name of environment variable
-     * @return value of environment variable or null if undefined
-     */
-    fun getenv(key: String): String?
-
+public interface PlatformEnvironProvider : EnvironmentProvider, PropertyProvider
+public interface PlatformProvider : PlatformEnvironProvider, Filesystem {
```

Comment:
**Question**: The `PlatformEnvironProvider` interface seems unnecessary. Why not just add `EnvironmentProvider` and `PropertyProvider` directly to `PlatformProvider`?

Reply:
Glad you asked (figured/hoped someone would).

That is how I had it originally and I back tracked a bit simply because smaller interfaces mean I can express what I depend on more precisely. An example is `AwsSdkSetting` which only needs `PlatformEnvironProvider` not all of the capabilities of `PlatformProvider`. 

That being said it's a minor thing and if you can make a case for just a single `PlatformProvider` interface that must implement all 3 then I'm happy to change it. If not and we want to keep it I'm open to hearing better names...not sure what to call something that provides environment variables and properties

Reply:
Yeah, this is a time where [intersection types](https://en.wikipedia.org/wiki/Intersection_type) would be really handy in the language.

I see what you're going for with minimal dependency scoping. I guess now, seeing how it's used in the companion PR, that I'd suggest calling this a `ConfigMapProvider` or a `PlatformSettingProvider` or something like that. Something that doesn't give undue weight to only one of the sub-interfaces.

Reply:
`EnvironmentPropertyFactorySettingPlatformProvider`?